lib/galaxy/jobs/__init__.py

```diff
@@ -171,54 +300,61 @@ def __init__(self, app):
 
         self.__parse_resource_parameters()
         # Initialize the config
-        job_config_file = self.app.config.job_config_file
         try:
-            tree = load(job_config_file)
-            self.__parse_job_conf_xml(tree)
+            if 'job_config' in self.app.config.config_dict:
+                job_config_dict = self.app.config.config_dict["job_config"]
+            else:
+                job_config_file = self.app.config.job_config_file
+                if '.xml' in job_config_file:
+                    tree = load(job_config_file)
+                    job_config_dict = self.__parse_job_conf_xml(tree)
+                else:
+                    import yaml
```

Comment:
That should move to the top I guess

Reply:
Rebased with this fix, thanks for the catch.